conans/util/windows.py

```diff
@@ -69,7 +71,9 @@ def path_shortener(path, short_paths):
         # cmd can fail if trying to set ACL in non NTFS drives, ignoring it.
         pass
 
-    redirect = tempfile.mkdtemp(dir=short_home, prefix="")
+    redirect = hashed_redirect(short_home, path) or tempfile.mkdtemp(dir=short_home, prefix="")
```

Comment:
I think this could hide some problems, like CI sharing a short-paths cache, and having multiple CI jobs start increasing the path lenghts, of the hashed version until failing (instead of having a different short-paths cache per CI job).

I think I'd prefer to raise if a hashed short path can't be found in the range.

Reply:
The fallback to `mkdtemp` was proposed by @jgsogo in this comment https://github.com/conan-io/conan/issues/3971#issuecomment-441200256

I'll let you decide what's the best choice here.  I'll adapt the PR when we reach consensus.

Reply:
I vote for raise + explanation in the exception for the poor user on how to recover, maybe just pointing to a section in the conan documentation ?

Reply:
I added a warning if the deterministic short path creation fails.